betfairlightweight/streaming/betfairstream.py

```diff
@@ -268,16 +268,28 @@ def __init__(self, directory, listener):
         """
         self.directory = directory
         self.listener = listener
+        self._running = False
 
     def start(self, async=False):
+        self._running = True
         if async:
             t = threading.Thread(name='HistoricalStream', target=self._read_loop)
             t.daemon = False
             t.start()
         else:
             self._read_loop()
 
+    def stop(self):
+        self._running = False
+
     def _read_loop(self):
-        with open(self.directory, 'r') as f:
-            for update in f:
-                self.listener.on_data(update)
+        while self._running:
```

Comment:
I think this needs to be moved into the for loop in some way as if you call stop() from outside the thread it won't actually stop the for loop

Reply:
Maybe a `if not self._running: break` after `self.listener.on_data(update)`

Reply:
OK!